js/signal_protocol_store.js

```diff
@@ -331,10 +331,10 @@
                     var promises = [];
                     while (sessions.length > 0) {
                         promises.push(new Promise(function(res) {
-                            sessions.pop().destroy().then(res);
+                            sessions.pop().destroy().then(res, res);
                         }));
                     }
-                    Promise.all(promises).then(resolve);
+                    Promise.all(promises).then(resolve, resolve);
```

Comment:
I can move it to `reject`, no problem. It is being called from a non-test location: `removeIdentityKey` in this same file.